core/modules/ip_checksum.cc

```diff
@@ -54,7 +56,8 @@ void IPChecksum::ProcessBatch(Context *ctx, bess::PacketBatch *batch) {
       data = qinq + 1;
       ether_type = qinq->ether_type;
       if (ether_type != be16_t(Ethernet::Type::kVlan)) {
-        continue;
+	EmitPacket(ctx, batch->pkts()[i], FORWARD_GATE);
+	continue;
```

Comment:
To be safe, could you include a note in the protobuf about how non-IP packets are simply forwarded with no modification.

also, nit: indentation got messed up here